File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/LongPropertySerializer.java
package ru.timeconqueror.timecore.common.capability.property.serializer;

import net.minecraft.nbt.CompoundTag;

public class LongPropertySerializer implements IPropertySerializer<Long> {
    public static final IPropertySerializer<Long> INSTANCE = new LongPropertySerializer();
    public static final IPropertySerializer<Long> NULLABLE = IPropertySerializer.nullable(INSTANCE);

    @Override
    public void serialize(String name, Long value, CompoundTag nbt) {
        nbt.putLong(name, value);
    }

    @Override
    public Long deserialize(String name, CompoundTag nbt) {
        return nbt.getLong(name);
    }
}

File: src/main/java/ru/timeconqueror/timecore/common/capability/property/serializer/LongPropertySerializer.kt
package ru.timeconqueror.timecore.common.capability.property.serializer

import net.minecraft.nbt.CompoundTag


object LongPropertySerializer : IPropertySerializer<Long> {

    override fun serialize(name: String, value: Long, nbt: CompoundTag) = nbt.putLong(name, value)
    override fun deserialize(name: String, nbt: CompoundTag) = nbt.getLong(name)

    object Nullable : NullPropertySerializer<Long>(this)
}